[Benjamin Koh Wei Jie] iP

Benjykoh opened the pull request:
# SQUISHYBOT

> "You're the bubble to my tea" - some guy on the internet

## Hello CU-TEA! This is Squishybot, the *FREE* friendly bubble tea chatbot :3

### Squishy will help you remember the things you have set out to accomplish! It is:
- Easy to pick up
- GUI Based
- Extremely fast

### To utilise Squishy, 
* Clone this [repo](https://github.com/Benjykoh/ip/)
* Run `Launcher.java` with any IDE
* Have fun!

### Launch Squishy here!
```ruby
public class Launcher {
    public static void main(String[] args) {
        Application.launch(Main.class, args);
}
```

### Squishy can:
- [X] Add tasks (todo, events, deadlines)
- [X] Delete them ⚔️ 
- [X] List them 📖 
- [X] Find them 🔍 

### **What are you waiting for? Squishy is already ready to be your companion!**

Commit:
Add Gradle support

Benjykoh committed:
Level-1

Commit:
Level-1

Benjykoh committed:
After Level-2

Benjykoh committed:
Implement Mark for Level-3

Benjykoh committed:
Implemented Unmark for Level-3

Benjykoh committed:
Implemented Deadline, Event and Todo for Level-4. Updated Parser and Printer

Benjykoh committed:
Implemented DukeException. Handled EmptyDescriptionException and UnknownCommandException.

Benjykoh committed:
Implemented InvalidInputException. Handled exceptions for mark, unmark, deadline, event and todo commands.

Benjykoh committed:
Implemented Delete for Level-6.

Benjykoh committed:
Updated Exception Handling

Benjykoh committed:
Updated Testcases

mradhyaf commented on src/main/java/Parser.java:
```diff
@@ -3,13 +3,19 @@
+        String[] array = input.split(" ");
+        String execute = array[0];
+        switch(execute) {
             case("list"):
```
`switch` and `case` statements should have the same level of indentation according to the [coding standard](https://se-education.org/guides/conventions/java/intermediate.html)

mradhyaf commented on src/main/java/TaskList.java:
```diff
@@ -1,21 +1,21 @@
+        taskList.add(task);
+    }
+
+    public int size() {
```
Perhaps a verb such as getSize would be better for the name of this method?

mradhyaf commented:
Almost LGTM, just a few nits to pick.

All the best for your iP!

Rye-Catcher commented on src/main/java/Parser.java:
```diff
@@ -0,0 +1,103 @@
+    public static void parse(String input, TaskList taskList) {
+        String[] array = input.split(" ");
+        String execute = array[0];
+        switch(execute) {
+            case("list"):
```
The switch statement should have the following form. There is no indentation for case clauses.

Rye-Catcher commented on src/main/java/Printer.java:
```diff
@@ -0,0 +1,86 @@
+                + "\n" + BLANK_LINE);
+    }
+
+    public static void todo(String todoString, TaskList taskList) {
+        String temp = "\n" + BLANK_LINE + "     Got it. I've added this task: \n" + "       [T][ ]" + todoString
+                + "\n" + "     Now you have " + taskList.size() + " task(s) in the list.\n" + BLANK_LINE;
+        System.out.println(temp);
+    }
+
+    public static void deadline(String deadlineName, String deadlineTime, TaskList taskList) {
+        String temp = "\n" + BLANK_LINE + "     Got it. I've added this task: \n" + "       [D][ ]" + deadlineName +
+                "(by:" + deadlineTime + ")" + "\n" +
+                "     Now you have " + taskList.size() + " task(s) in the list.\n" + BLANK_LINE;
+        System.out.println(temp);
+    }
+
+    public static void event(String eventName, String eventTime, TaskList taskList) {
+        String temp = "\n" + BLANK_LINE + "     Got it. I've added this task: \n" + "       [E][ ]" + eventName +
+                "(at: " + eventTime + ")" + "\n" +
+                "     Now you have " + taskList.size() + " task(s) in the list.\n" + BLANK_LINE;
+        System.out.println(temp);
+    }
+
+    public static void delete(int numToDelete, TaskList taskList) {
```
Since this method has a wide scope. Maybe it is better to have a name with a more direct meaning? Like `printTodo` e.t.c.

Rye-Catcher reviewed (commented):
Overall the code looks good. 👍  But many public methods are not commented using Java Doc style. Also, there are a few layout style violations.

Benjykoh committed:
Implemented Save for Level 7

Benjykoh committed:
Updated Save for Level-7

Benjykoh committed:
Update for Level-7

Benjykoh committed:
Updated for Level-8

Benjykoh committed:
Merge branch 'branch-Level-8'

* branch-Level-8:
  Updated for Level-8

# Conflicts:
#	src/main/java/Parser.java
#	src/main/java/Ui.java

Benjykoh committed:
Resolved Merge Conflicts

Benjykoh committed:
Used more OOP and resolved merge conflicts

Benjykoh committed:
A-Packages

Benjykoh committed:
Added JUnit Tests

Benjykoh committed:
Updated Test Cases

Benjykoh committed:
Added JavaDocs comments

Benjykoh committed:
Tweaked code to comply with coding standard

Benjykoh committed:
Implemented Find for Level-9

Benjykoh committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/Event.java
#	src/main/java/duke/TaskList.java
#	src/main/java/duke/Ui.java

Benjykoh committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/Parser.java

Benjykoh committed:
A-Jar

Benjykoh committed:
Merge branch 'add-gradle-support'

Benjykoh committed:
Updated Dependencies

Benjykoh committed:
Implement GUI for Level-10

Benjykoh committed:
Added new files for GUI

Benjykoh committed:
Edited UI

Benjykoh committed:
Update Duke.java

Duke.java does not ensure there is a input.

An assertion will ensure the user will always supply an input.

Let's,
*Add an assertion to make sure that there will always be an input.

Benjykoh committed:
Update Duke.java

Duke: Ensure there will always be an user input

getResponse() might not always receive an input

Adding an assertion will ensure the user will always supply an
input.

Let's:
*Add an assertion into getResponse()

Benjykoh committed:
Merge pull request #2 from Benjykoh/heads/A-Assertions

Update Duke.java

Benjykoh committed:
Remove dead code

Multiple classes contain dead code.

Dead code makes it harder to understand code and takes up
redundant space.

Let's remove dead code in all relevant classes.

Benjykoh committed:
Merge pull request #3 from Benjykoh/A-CodeQuality

Remove dead code

Benjykoh committed:
Update loadfile()

loadfile() did not mark tasks as done. If task was done, it was
not reflected as done when a new instance of Launcher was launched.

Let's update the method to ensure tasks will be marked as done.

Benjykoh committed:
Update GUI

GUI lacked personality.

Let's,
 * Added new user image and bot image.
 * Add a new font

Benjykoh committed:
Add resources for GUI

Add files for the new user and bot images.
Add new file for font.

Benjykoh committed:
Add Ui.png

Add screenshot of Ui

Benjykoh committed:
Update README.md

Benjykoh committed:
Update Parser and UI

Find method return entire list

UI did not handle empty lists or unable to find task

Let's
* Update find method to return appropriately
* Update UI to handle more cases

Benjykoh committed:
Set theme jekyll-theme-architect

Benjykoh committed:
Update Name
Update UserGuide

Benjykoh committed:
Improve abstraction and Fix bugs

Reduced length of methods in Parser.java.

Handled more Exceptions in Mark, Unmark and Todo commands.

Benjykoh committed:
Update Checkstyle

Corrected checkstyle to fit the correct format.

Benjykoh committed:
Abstract details out of certain methods

For parser, exceptions were abstracted out.

For storage, logic of input handler was abstracted out.
